sock.c

```diff
@@ -25,130 +25,71 @@
 
 #include "common.h"
 #include "sock.h"
-#include <arpa/inet.h>
 
 #define RETRACE_MAX_IP_ADDR_LEN 15
+
+int RETRACE_IMPLEMENTATION(socket)(int domain, int type, int protocol)
+{
+	int sock;
+	rtr_socket_t real_socket;
+
+	real_socket = RETRACE_GET_REAL(socket);
+
+	sock = real_socket(domain, type, protocol);
+	trace_printf(1, "socket(%d, %d, %d) [return: %d]\n", domain, type, protocol, sock);
+
+	return sock;
+}
+
+RETRACE_REPLACE(socket)
+
 #ifdef __linux__
 int RETRACE_IMPLEMENTATION(connect)(int fd, __CONST_SOCKADDR_ARG _address, socklen_t len)
 #else
 int RETRACE_IMPLEMENTATION(connect)(int fd, const struct sockaddr *address, socklen_t len)
 #endif
 {
-	unsigned short port;
-	int match_port;
-	int redirect_port;
-	char *match_ip = NULL;
-	char *redirect_ip = NULL;
 	rtr_connect_t real_connect;
-	char ip_address[RETRACE_MAX_IP_ADDR_LEN + 1];
 #ifdef __linux__
 	const struct sockaddr *address = _address.__sockaddr__;
 #endif
 
-	port = ntohs(*(unsigned short *)&address->sa_data[0]);
+	int ret;
 
 	real_connect = RETRACE_GET_REAL(connect);
 
-	/* Only implemented for IPv4 right now */
-	if (get_tracing_enabled() && address->sa_family == AF_INET) {
-		FILE *config = NULL;
-
-		while (1) {
-			int ret;
-			struct sockaddr match_addr;
-
-			ret = rtr_get_config_multiple(&config, "connect",
```

Comment:
@riataman I'm going to add it after added socket API functions such as send, recv, sendfrom, recvfrom, setsocketopt, ioctl...